pkgs/development/tools/poac/default.nix

```diff
@@ -0,0 +1,129 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, cpm
+, git
+, cacert
+, boost179
+, fmt_8
+, icu
+, libarchive
+, libgit2
+, lz4
+, openssl_3
+, spdlog
+}:
+
+let
+  git2Cpp = fetchFromGitHub {
+    owner = "ken-matsui";
+    repo = "git2-cpp";
+    rev = "v0.1.0-alpha.1";
+    sha256 = "sha256-Ub0wrBK5oMfWGv+kpq/W1PN3yzpcfg+XWRFF/lV9VCY=";
+  };
+
+  glob = fetchFromGitHub {
+    owner = "p-ranav";
+    repo = "glob";
+    rev = "v0.0.1";
+    sha256 = "sha256-2y+a7YFBiYX8wbwCCWw1Cm+SFoXGB3ZxLPi/QdZhcdw=";
+  };
+
+  packageProjectCMake = fetchFromGitHub {
+    owner = "TheLartians";
+    repo = "PackageProject.cmake";
+    rev = "v1.3";
+    sha256 = "sha256-ZktftDrPo+JhBt0XKJekv0cyxIagvcgMcXZOBd4RtKs=";
+  };
+
+  mitamaCppResult = fetchFromGitHub {
+    owner = "LoliGothick";
+    repo = "mitama-cpp-result";
+    rev = "v9.3.0";
+    sha256 = "sha256-CWYVPpmPIZZTsqXKh+Ft3SlQ4C9yjUof1mJ8Acn5kmM=";
+  };
+
+  ninja = fetchFromGitHub {
```

Comment:
Now I'm a bit afraid. There is a huge vendoring occurring here.

Reply:
@AndersonTorres 

https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/build-managers/ninja/default.nix

I need the source code of ninja, but does this provide (keep) the source code?

Reply:
Well, I don't know precisely what you want.
If you need the raw source code, you can e.g. use `ninja.src`. 
However, if by any reason you need a particular source that is not provided by us (say, a bleeding edge, not-yet-released ninja), you can do as you did here: `fetchFromGitHub {...}`.

I am more concerned about how this software is made. Using vendoring and bundling is a bit unusual, and, [according to some manuals](https://gist.github.com/datagrok/8577287), [a bad practice](https://blogs.gentoo.org/mgorny/2021/02/19/the-modern-packagers-security-nightmare/).

(Technically, this is not vendoring, because vendoring is (by definition) keeping a copy of the extra code, whereas you are keeping a link plus a recipe of how to retrieve it.)

But this is not my concern, indeed. Do as you was doing before.

Reply:
@AndersonTorres 

I see. `ninja` is the only library that is static-linked. All other libraries are header-only, so no link occurs.

Therefore, if I understand what you mean correctly, only `ninja` fits the bad practice you shared.

That being said, the new version of `ninja` has been released, so I think I can replace it.

> But this is not my concern, indeed. Do as you was doing before.

So do I not need to create nixpkgs of them? I thought vendoring meant the use of `fetchFromGitHub`, but it seems not.

Reply:
> So do I not need to create nixpkgs of them? I thought vendoring meant the use of fetchFromGitHub, but it seems not.

The term is confusing, indeed. Forget it for now.

As I have noticed, the projects you are downloading via `fetch...` are independent, and some of them are not in Nixpkgs yet. So, as a side issue, it is desirable to include them in Nixpkgs later in the future.

On the other hand, you are using them as source files, not as prebuilt `binaryNativeCode`. This is not so much a problem for Nixpkgs side, because Nixpkgs infrastructure merely downloads and provides them.

This is more of a problem for the upstream maintainers of poac itself. However I am not a developer of poac, it is not my duty to tell them how they should develop their project.

Reply:
@AndersonTorres 

I'm a core maintainer of Poac, and I actually don't care the `binaryNativeCode` and other things for now. I'll have to work on simplifying the treatment of libraries. I'd like to link Poac with libraries dynamically but not all libraries do not provide linkable libraries. For example, Ninja is not intended to be used as a library so it doesn't have public headers and libraries. That's why we need to build it from the source. While Poac depends on many libraries, it's hard to make them ideal. That is the problem with C++ in that each library has various ways of providing itself. With Rust, that should not be a problem thanks to Cargo. Thus, in terms of development priorities, I will be solving this problem in the distant future, not by doing it now.

Do you have any other concerns you cannot still accept this PR?

Reply:
Understood.  There is only two more things to fix.

Reply:
Thank you for your review.